pkg/processor/eventprocessor/pool.go

```diff
@@ -237,3 +237,38 @@ func (nba *nonBlockingPoolAllocator) SetObjects(objects []EventProcessor) error
 	nba.logger.DebugWith("Allocator objects updated", "size", len(objects))
 	return nil
 }
+
+// GetStatistics returns aggregated allocation statistics from all objects in the pool.
+// The non-blocking allocator itself does not maintain internal statistics,
+// as it performs allocations in a non-blocking manner.
+// Therefore, it delegates statistics collection to the individual objects it manages.
+func (nba *nonBlockingPoolAllocator) GetStatistics() *statistics.AllocatorStatistics {
+	// Initialize a struct to accumulate total statistics
+	totalStats := &statistics.AllocatorStatistics{}
+	numProcessors := 0
+	var percentageSum uint64
+
+	for _, object := range nba.objects {
+		stats := object.GetAllocationStatistics()
```

Comment:
Is this function called in a critical path? If so, we might consider to do this concurrently.

Reply:
@TomerShor no, it's not a critical path. And since GetAllocationStatistics() is lightweight, introducing concurrency might only add unnecessary overhead and complexity.